contracts/multisig-prover/src/encoding/rkyv/mod.rs

```diff
@@ -0,0 +1,157 @@
+use axelar_rkyv_encoding::types::{ECDSA_COMPRESSED_PUBKEY_LEN, ED25519_PUBKEY_LEN};
+use itertools::Itertools;
+use multisig::{
+    key::{PublicKey, Recoverable, Signature},
+    msg::SignerWithSig,
+    verifier_set::VerifierSet,
+};
+use router_api::Message;
+use std::{array::TryFromSliceError, collections::BTreeMap};
+
+use crate::{error::ContractError, payload::Payload};
+
+type Result<T> = core::result::Result<T, ContractError>;
+
+pub fn to_verifier_set(vs: &VerifierSet) -> Result<axelar_rkyv_encoding::types::VerifierSet> {
+    let mut signers: BTreeMap<
+        axelar_rkyv_encoding::types::PublicKey,
+        axelar_rkyv_encoding::types::U256,
+    > = BTreeMap::new();
+
+    vs.signers
+        .iter()
+        .try_for_each(|(_, signer)| -> Result<()> {
+            let enc_pubkey = to_pub_key(&signer.pub_key)?;
+            let enc_weight =
+                axelar_rkyv_encoding::types::U256::from_le(to_u256_le(signer.weight.u128()));
+
+            signers.insert(enc_pubkey, enc_weight);
```

Comment:
@roberts-pumpurs , @tilacog , please let's ensure this is correct. I am directly using `signer.pub_key` here, and avoiding a conversion from a String (the keys of the map), assuming they are the same.